File: C#/EntityFramework_ORM/DAL/Entities/Order.cs
﻿using DAL.Interfaces;
using System;
using System.Collections.Generic;

namespace MyApp.DAL.Entities;

public partial class Order : IEntity
{
    public int OrderID { get; set; }

    public string? CustomerID { get; set; }

    public int? EmployeeID { get; set; }

    public DateTime? OrderDate { get; set; }

    public DateTime? RequiredDate { get; set; }

    public DateTime? ShippedDate { get; set; }

    public int? ShipVia { get; set; }

    public decimal? Freight { get; set; }

    public string? ShipName { get; set; }

    public string? ShipAddress { get; set; }

    public string? ShipCity { get; set; }

    public string? ShipRegion { get; set; }

    public string? ShipPostalCode { get; set; }

    public string? ShipCountry { get; set; }

    public virtual Customer? Customer { get; set; }

    public virtual Employee? Employee { get; set; }

    public virtual ICollection<Order_Detail> Order_Details { get; set; } = new List<Order_Detail>();

    public virtual Shipper? ShipViaNavigation { get; set; }
}

File: Go/SQLBoiler_ORM/dal/entities/order.go
package entities

import "time"

type Order struct {
	OrderID        int        `db:"OrderID"`
	CustomerID     *string    `db:"CustomerID"`
	EmployeeID     *int       `db:"EmployeeID"`
	OrderDate      *time.Time `db:"OrderDate"`
	RequiredDate   *time.Time `db:"RequiredDate"`
	ShippedDate    *time.Time `db:"ShippedDate"`
	ShipVia        *int       `db:"ShipVia"`
	Freight        *float64   `db:"Freight"`
	ShipName       *string    `db:"ShipName"`
	ShipAddress    *string    `db:"ShipAddress"`
	ShipCity       *string    `db:"ShipCity"`
	ShipRegion     *string    `db:"ShipRegion"`
	ShipPostalCode *string    `db:"ShipPostalCode"`
	ShipCountry    *string    `db:"ShipCountry"`
}
